internal/driver/nodeserver_helpers.go

```diff
@@ -328,39 +328,24 @@ func (ns *NodeServer) mountVolume(ctx context.Context, devicePath string, req *c
 // If not, it formats the device using the provided LuksContext.
 // Finally, it prepares the LUKS volume for mounting.
 func (ns *NodeServer) prepareLUKSVolume(ctx context.Context, devicePath string, luksContext LuksContext) (string, error) {
+	var luksSource string
+	var err error
 	log := logger.GetLogger(ctx)
 	log.V(4).Info("Entering prepareLUKSVolume", "devicePath", devicePath, "luksContext", luksContext)
 
 	// LUKS encryption enabled, check if the volume needs to be formatted.
 	log.V(4).Info("LUKS encryption enabled")
 
-	// Validate if the device is formatted with LUKS encryption or if it needs formatting.
-	formatted, err := ns.encrypt.blkidValid(devicePath)
-	if err != nil {
-		return "", errInternal("Failed to validate blkid (%q): %v", devicePath, err)
-	}
-	log.V(4).Info("Disk format check", "devicePath", devicePath, "formatted", formatted)
```

Comment:
question: with new luks implementation, do we not need to check if the device is formatted already?

Reply:
added a check now